src/Controller/ExportController.php

```diff
@@ -29,24 +31,48 @@ class ExportController extends AppController
     const DEFAULT_EXPORT_LIMIT = 10000;
 
     /**
-     * Export data in csv format
+     * {@inheritDoc}
+     * {@codeCoverageIgnore}
+     */
+    public function initialize(): void
+    {
+        parent::initialize();
+
+        $this->loadComponent('Export');
+    }
+
+    /**
+     * Export data to format specified by user
      *
-     * @return \Cake\Http\Response
+     * @return \Cake\Http\Response|null
      */
-    public function export(): Response
+    public function export(): ?Response
     {
+        // allow only specific export formats: csv, ods, xlsx
+        $format = (string)$this->request->getData('format');
+        if (!in_array($format, ['csv', 'ods', 'xlsx'])) {
+            return null;
+        }
         // check request (allowed methods and required parameters)
         $data = $this->checkRequest([
             'allowedMethods' => ['post'],
             'requiredParameters' => ['objectType'],
         ]);
         $ids = $this->request->getData('ids');
 
-        // load csv data for objects by object type and ids
-        $rows = $this->csvRows($data['objectType'], $ids);
+        // load data for objects by object type and ids
+        $rows = $this->rows($data['objectType'], $ids);
+
+        // create spreadsheet and return as download
+        $spreadsheet = $this->Export->spreadsheet($rows);
+        $filename = sprintf('%s_%s.%s', $data['objectType'], date('Ymd-His'), $format);
+        $data = $this->Export->{$format}($spreadsheet, $filename);
```

Comment:
I'd prefer an `ExportComponent::format($format, $rows, $filename)` and no `public` `spreadsheet()` method

 * in `ExportComponent::format()` we could load other formats via plugin not requiring `spreadsheet()` method
 * we could simply add a `json` format easily 
 
Also an `ExportComponent::checkFormat()` could be used, only `ExportComponent` knows the available formats?
Or an `ExportComponent::ALLOWED_FORMATS` array for now.

Reply:
ok. some refactor soon